vortex-layout/src/layouts/struct_/reader.rs

```diff
@@ -174,19 +178,26 @@ impl LayoutReader for StructReader {
         Precision::Exact(self.layout.row_count())
     }
 
-    fn register_splits(
-        &self,
-        field_mask: &[FieldMask],
-        row_offset: u64,
-        splits: &mut BTreeSet<u64>,
-    ) -> VortexResult<()> {
-        // In the case of an empty struct, we need to register the end split.
-        splits.insert(row_offset + self.layout.row_count);
-
-        self.layout.matching_fields(field_mask, |mask, idx| {
-            self.child_by_idx(idx)?
-                .register_splits(&[mask], row_offset, splits)
-        })
+    fn row_masks(&self, selection: &TreeRowMask, field_mask: &[FieldMask]) -> MaskStream {
+        // Here we construct a stream of masks for each field in the field_mask, and then take
+        // the smallest mask from each field.
+        // If the field_mask is empty, we return a stream of all true masks.
+        if field_mask.is_empty() {
+            let row_count = self.layout.row_count().as_usize();
```

Comment:
You may just want to implement `all_true_mask_stream(u64)` that returns multiple masks if n > u32::max.

Reply:
You should abstract your all_false mask stream into a constant_mask_iterator!